bhttp/src/http_compat/encode.rs

```diff
@@ -0,0 +1,431 @@
+    /// * `Poll::Ready(None)` - All data has been encoded
+    /// * `Poll::Pending` - No data is ready yet, but may be in the future
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+
+        match &mut this.state {
+            EncodeState::ControlAndHeaders => {
+                let mut buf = Vec::new();
+
+                let control_data = this.message.bhttp_control_data()?;
+
+                // 1. Write the framing indicator.
+                control_data
+                    .framing_indicator(Mode::IndeterminateLength)
+                    .write_bhttp(&mut buf)?;
+
+                // 2. Response with with informational code (100 - 199) is not supported so we just skip encoding Informational Response.
+                if let Some(code) = control_data.status() {
+                    if code.informational() {
+                        return Poll::Ready(Some(Err(Error::UnsportedStatusCode(code.code()))));
+                    }
+                }
+
+                // 3. Write (request / the final response) control data
+                control_data.write_bhttp(&mut buf)?;
+
+                // 4. Write headers field section
+                let field_section = this.message.bhttp_headers()?;
+                field_section.write_bhttp(Mode::IndeterminateLength, &mut buf)?;
+
+                // 5. Change to next state
+                this.state = EncodeState::BodyChunk {
+                    collected_trailers: None,
+                };
+                Poll::Ready(Some(Ok(buf)))
+            }
+            EncodeState::BodyChunk { collected_trailers } => {
+                // Poll the body for more data
+                let body = this.message.body_mut();
+                match Pin::new(body).poll_frame(cx) {
+                    Poll::Ready(Some(Ok(frame))) => {
+                        match frame.into_data() {
+                            Ok(data) => {
+                                // Is a data frame
+                                let mut chunk_data = Vec::new();
+                                write_vec(&data, &mut chunk_data)?;
+                                Poll::Ready(Some(Ok(chunk_data)))
+                            }
+                            Err(frame) => {
+                                if let Ok(trailers) = frame.into_trailers() {
+                                    // Is a trailers frame
+                                    if let Some(current) = collected_trailers {
+                                        current.extend(trailers);
+                                    } else {
+                                        *collected_trailers = Some(trailers);
+                                    }
+                                    Poll::Pending
+                                } else {
+                                    // For unknown frame types, report an invalid data error.
```

Comment:
That's not quite right.  If this is HTTP/2, then you are required to ignore unknown frames.